src/contrib/dependencyinjection/Akka.DependencyInjection.Tests/ActorServiceProviderPropsWithScopesSpecs.cs

```diff
@@ -0,0 +1,283 @@
+                Dependencies = dependencies;
+            }
+
+            public AkkaDiFixture.IDependency[] Dependencies { get; }
+        }
+
+        public class ScopedActor : ReceiveActor
+        {
+            private readonly IServiceScope _scope;
+            private AkkaDiFixture.ITransientDependency _transient;
+            private AkkaDiFixture.IScopedDependency _scoped;
+
+            public ScopedActor(IServiceProvider sp)
+            {
+                _scope = sp.CreateScope();
+
+                Receive<FetchDependencies>(_ =>
+                {
+                    Sender.Tell(new CurrentDependencies(new AkkaDiFixture.IDependency[] { _transient, _scoped }));
+                });
+
+                Receive<Crash>(_ => throw new ApplicationException("crash"));
+            }
+
+            protected override void PreStart()
+            {
+                _scoped = _scope.ServiceProvider.GetService<AkkaDiFixture.IScopedDependency>();
+                _transient = _scope.ServiceProvider.GetRequiredService<AkkaDiFixture.ITransientDependency>();
+            }
+
+            protected override void PostStop()
+            {
+                _scope.Dispose();
+            }
+        }
+
+        public class MixedActor : ReceiveActor
+        {
+            private readonly AkkaDiFixture.ISingletonDependency _singleton;
+            private readonly IServiceScope _scope;
+            private AkkaDiFixture.ITransientDependency _transient;
+            private AkkaDiFixture.IScopedDependency _scoped;
+
+            public MixedActor(AkkaDiFixture.ISingletonDependency singleton, IServiceProvider sp)
+            {
+                _singleton = singleton;
+                _scope = sp.CreateScope();
+
+                Receive<FetchDependencies>(_ =>
+                {
+                    Sender.Tell(new CurrentDependencies(new AkkaDiFixture.IDependency[] { _transient, _scoped, _singleton }));
+                });
+
+                Receive<Crash>(_ => throw new ApplicationException("crash"));
+            }
+
+            protected override void PreStart()
+            {
+                _scoped = _scope.ServiceProvider.GetService<AkkaDiFixture.IScopedDependency>();
+                _transient = _scope.ServiceProvider.GetRequiredService<AkkaDiFixture.ITransientDependency>();
+            }
+
+            protected override void PostStop()
+            {
+                _scope.Dispose();
+            }
+        }
+
+        public class NonDiArgsActor : ReceiveActor
+        {
+            private readonly AkkaDiFixture.ISingletonDependency _singleton;
+            private readonly IServiceScope _scope;
+            private AkkaDiFixture.ITransientDependency _transient;
+            private AkkaDiFixture.IScopedDependency _scoped;
+            private string _arg1;
+            private string _arg2;
+
+            public NonDiArgsActor(AkkaDiFixture.ISingletonDependency singleton, IServiceProvider sp, string arg1, string arg2)
```

Comment:
Using this to test out whether or not we can mix DI and constructor arguments provided directly via the `ServiceProvider.Props`.